### PR TITLE
35 us 030101

### LINKED ISSUE #35: US 03.01.01
As an administrator, I want to be able to remove events.

### PR DESCRIPTION
**Pull request by Jameszf**
# Code
 - Add deleting event functionality.
 - Connected PostEvent to Firestore.
 - Connected AllEvents to Firestore.

https://github.com/user-attachments/assets/e58b517a-cc56-4dba-a324-f4056b7ef486

# Other
 - CRC and UML updated.
 - No intent tests need to be made because no new fragment was added.
 - No unit tests were made.

### CODE REVIEW

File: app/src/main/java/com/example/pygmyhippo/user/EventFragment.java [hunk 1]
```diff
@@ -90,57 +191,56 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         eventAboutDescriptionView.setText(event.getDescription());
     }
 
-    @Override
```
Comment (katzhong) [inline app/src/main/java/com/example/pygmyhippo/user/EventFragment.java:104]: just curious, why did you remove oncreateview here?

**Comment by katzhong** [comment]
very nice, added one comment, but otherwise, looks good

File: app/src/main/java/com/example/pygmyhippo/user/EventFragment.java (hunk 1 above)
Comment (Jameszf) [reply app/src/main/java/com/example/pygmyhippo/user/EventFragment.java:104]: OCD. Its still in EventFragment just at a different spot in the file.

File: app/src/androidTest/java/com/example/pygmyhippo/organiser/PostEventFragmentTesting.java [hunk 2]
```diff
@@ -1,18 +1,14 @@
 package com.example.pygmyhippo.organiser;
-import com.example.pygmyhippo.R;
+
 import static androidx.test.espresso.Espresso.onView;
-import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.action.ViewActions.click;
```
Comment (Kori-Kozicki) [inline app/src/androidTest/java/com/example/pygmyhippo/organiser/PostEventFragmentTesting.java]: Is this the import that fixes the UI test bug?

File: app/src/main/java/com/example/pygmyhippo/MainActivity.java [hunk 3]
```diff
@@ -198,6 +201,27 @@ private void setupNavControllerOrganiser(BottomNavigationView navView, Bundle bu
...
+                navController.navigate(R.id.admin_navigation_all_events, bundle);
+            } else if (item.getItemId() == R.id.admin_all_images_menu_item) {
+                navController.navigate(R.id.admin_navigation_all_images, bundle);
+            } else if (item.getItemId() == R.id.admin_all_profiles_menu_item) {
+                navController.navigate(R.id.admin_navigation_all_users, bundle);
+            }
+            return true;
+        });
+    }
+
```
Comment (Kori-Kozicki) [inline app/src/main/java/com/example/pygmyhippo/MainActivity.java:224]: Did this method ever get a JavaDoc?

File: app/src/main/java/com/example/pygmyhippo/user/EventFragment.java [hunk 4]
```diff
@@ -63,24 +80,108 @@ public Event hardcodeEvent() {
...
+                Log.d("EventFragment", "Successfully deleted event. Navigating back to all events");
+                Bundle navArgs = new Bundle();
+                navArgs.putParcelable("signedInAccount", signedInAccount);
+                navController.navigate(R.id.admin_navigation_all_events, navArgs);
+            } else {
+                Log.d("EventFragment", "Error in deleting event.");
+            }
+        }
+    }
+
```
Comment (Kori-Kozicki) [inline app/src/main/java/com/example/pygmyhippo/user/EventFragment.java:174]: Just wanna stay up to speed here, does this populate the event fragment texts with an event from the database?

File: app/src/main/java/com/example/pygmyhippo/user/EventFragment.java [hunk 5]
```diff
@@ -63,24 +80,108 @@ public Event hardcodeEvent() {
...
+    private void getEvent(@Nullable String eventID) {
+        if (eventID == null) {
+            Log.d("EventFragment", "No Event ID was passed via navigation to EventFragment, using mock data...");
+            event = hardcodeEvent();
+            populateTextFields();
+        } else {
+            Log.d("EventFragment", String.format("Non-null eventID, attempting to retrieve Event with ID %s", eventID));
+            handler.getEventByID(eventID, this);
+        }
+    }
```
Comment (Kori-Kozicki) [inline app/src/main/java/com/example/pygmyhippo/user/EventFragment.java:190]: Oooo this is good for testing. In the future are you planning to put something else here if the event isn't found? Like something to indicate database error maybe

File: app/src/main/java/com/example/pygmyhippo/user/EventFragment.java [hunk 6]
```diff
@@ -90,57 +191,56 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+    private void setPermissions() {
+        if (signedInAccount.getCurrentRole() == Account.AccountRole.admin) {
+            Log.d("EventFragment", "Admin user detected. Setting Admin permissions");
+            adminConstraint.setVisibility(View.VISIBLE);
+            registerButton.setVisibility(View.GONE);
+        } else {
+            Log.d("EventFragment", "Non-Admin user detected. Setting non-Admin permissions");
+            adminConstraint.setVisibility(View.GONE);
+            registerButton.setVisibility(View.VISIBLE);
+        }
```
Comment (Kori-Kozicki) [inline app/src/main/java/com/example/pygmyhippo/user/EventFragment.java:255]: Wait so the event view from admin just uses the same as the user? That's pretty nice. Maybe just update in the header comment that Admin role uses this too just to make it clear

**Review by Kori-Kozicki — APPROVED** [review]
Firstly can I just say how much I love that you included the demo video. Makes it so much easier to review :)

Overall I think this is really good. I like that you added more database functionality.  I see you have some image stuff here too. Does that mean this PR also solves the browsing images user story? Because if it does it's not listed in the PR

**Review by SuperCreativeUniqueUsername — APPROVED** [review]
